### PR TITLE
Additional cli options for printing AST's and running single rules

### PR DESCRIPTION
**Pull request by arturbosch**
- flag `--print-ast` helps to debug code while writing new rules
- flag `--run-rule` allows to run single (new) rules on for example our analyzing projects pipeline #846 

Todo:
- tests
- handle MultiRule `ruleSet:rule:subrule`
- handle config properties `ruleSet:rule[config=properties]`

### CODE REVIEW

File: detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Args.kt [hunk 1]
```diff
@@ -62,13 +62,21 @@ class Args {
...
 	@Parameter(names = ["--help", "-h"],
 			help = true, description = "Shows the usage.")
 	var help: Boolean = false
 
+	@Parameter(names = ["--run-rule"],
+			description = "Specify a rule by [RuleSet:Rule] pattern and run it on input.")
+	var runRule: String? = null
+
+	@Parameter(names = ["--print-ast"],
+			description = "Prints the AST for given [input] file. Must be no directory.")
```
Comment (schalkms) [inline detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Args.kt]: *Must be no directory* -> Must be a file? would be more clear IMO

Comment (arturbosch) [reply detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Args.kt]: Sure will add them

Comment (arturbosch) [reply detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Args.kt]: test cases added!

File: detekt-cli/src/test/resources/cases/Poko.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package cases
+
+class Poko {
+
+	val name: String = "Poko"
```
Comment (schalkms) [inline detekt-cli/src/test/resources/cases/Poko.kt]: Interesting name ;)

**Pull request merged**